lib/active_job/failed.rb

```diff
@@ -4,8 +4,4 @@ module ActiveJob::Failed
   included do
     attr_accessor :last_execution_error, :failed_at
   end
-
-  def failed?
-    last_execution_error.present?
-  end
```

Comment:
I think this might potentially cause an infinite recursion problem with this method: 
https://github.com/basecamp/mission_control-jobs/blob/6d7315b159a33e8b2c4261f322841c4324d41729/lib/active_job/executing.rb#L28-L32

Together with your change a bit below to do 
```ruby
 ActiveJob::JobsRelation::STATUSES.each do |status|
    define_method "#{status}?" do
      self.status == status
    end
  end
```
if `@status` is not defined, as we'd call `failed?`, that will try to call `self.status`, which will call `failed?` again and so on. 

This was done to support Resque and Solid Queue together. If we can guarantee that `@status` will always be defined, then we should just remove the `status` method and rely only on the attribute, which should change from `attr_writer` to `attr_accessor`. I think we can guarantee that `@status` will be set now 🤔  Initially when I modified several parts of Mission Control to work with jobs in different statuses other than `failed` and `pending`, and `failed` jobs that could be in a queue, this wasn't the case, but I think I covered all the places where we instantiate `JobProxy` and there's no other way to access a job within Mission Control.

Reply:
Got it, that's really helpful.
Thank you!